clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusClusterCachingAgent.groovy

```diff
@@ -83,6 +98,87 @@ class TitanClusterCachingAgent implements CachingAgent { //, OnDemandAgent {
     types
   }
 
+  @Override
+  String getOnDemandAgentType() {
+    "${getAgentType()}-OnDemand"
+  }
+
+  @Override
+  OnDemandMetricsSupport getMetricsSupport() {
+    return metricsSupport
+  }
+
+  @Override
+  boolean handles(String type) {
+    return false
+  }
+
+  @Override
+  boolean handles(String type, String cloudProvider) {
+    return type == ON_DEMAND_TYPE && cloudProvider == titusCloudProvider.id
+  }
+
+  @Override
+  OnDemandAgent.OnDemandResult handle(ProviderCache providerCache, Map<String, ? extends Object> data) {
+    if (!data.containsKey("serverGroupName")) {
+      return null
+    }
+    if (!data.containsKey("account")) {
+      return null
+    }
+    if (!data.containsKey("region")) {
+      return null
+    }
+
+    if (account.name != data.account) {
+      return null
+    }
+
+    if (region != data.region) {
+      return null
+    }
+
+    //TODO(cfieber) - this should just load a single server group and follow the same on-demand caching behaviour as the AWS provider
+    List<Job> jobs = metricsSupport.readData {
+      titusClient.getAllJobs()
+    }
+
+    CacheResult result = metricsSupport.transformData { buildCacheResult(jobs) }
+    def withOnDemand = new DefaultCacheResult(result.cacheResults + [onDemand: [new DefaultCacheData(
+      Keys.getServerGroupKey(data.serverGroupName, data.account, data.region),
+      10 * 60, // ttl is 10 minutes
+      [
+        cacheTime: System.currentTimeMillis(),
+        cacheResults: "[{}]",
+        processedCount: 1,
+        processedTime: System.currentTimeMillis()
+      ],
+      [:]
+    )]], result.evictions)
+    new OnDemandAgent.OnDemandResult(sourceAgentType: getAgentType(), authoritativeTypes: [SERVER_GROUPS.ns], cacheResult: withOnDemand)
+  }
+
+  @Override
+  Collection<Map> pendingOnDemandRequests(ProviderCache providerCache) {
+    def keys = providerCache.getIdentifiers('onDemand')
+    keys = keys.findResults {
+      def key = Keys.parse(it)
```

Comment:
@ajordens and @lwander I think the ON_DEMAND stuff will have some problems in a multi-provider world. At the very least each implementation needs to ensure it is only dealing with its own keys (Keys.parse should return null if the key isn't prefixed with the provider id)

I think this is broken in `ClusterCachingAgent` not sure if it is in `KubernetesServerGroupCachingAgent` or not